modules/gdscript/gdscript_parser.cpp

```diff
@@ -489,14 +583,27 @@ void GDScriptParser::parse_program() {
 	head->fqcn = script_path;
 	current_class = head;
 	bool can_have_class_or_extends = true;
+	Vector<String> top_header_comment;
 
 	while (!check(GDScriptTokenizer::Token::TK_EOF)) {
-		if (match(GDScriptTokenizer::Token::ANNOTATION)) {
+		if (check(GDScriptTokenizer::Token::COMMENT)) {
+			top_header_comment = check_for_comment_block();
+		} else if (match(GDScriptTokenizer::Token::ANNOTATION)) {
 			AnnotationNode *annotation = parse_annotation(AnnotationInfo::SCRIPT | AnnotationInfo::STANDALONE | AnnotationInfo::CLASS_LEVEL);
 			if (annotation != nullptr) {
 				if (annotation->applies_to(AnnotationInfo::SCRIPT)) {
-					// `@icon` needs to be applied in the parser. See GH-72444.
+					// `@icon` and `@tool` need to be applied in the parser. See GH-72444.
 					if (annotation->name == SNAME("@icon")) {
+						head->icon_inline_comment = check_for_comment();
+						match(GDScriptTokenizer::Token::NEWLINE);
+						head->icon_header_comment = top_header_comment.duplicate();
+						top_header_comment.clear();
+						annotation->apply(this, head);
+					} else if (annotation->name == SNAME("@tool")) {
+						head->tool_inline_comment = check_for_comment();
+						match(GDScriptTokenizer::Token::NEWLINE);
+						head->tool_header_comment = top_header_comment.duplicate();
+						top_header_comment.clear();
 						annotation->apply(this, head);
```

Comment:
Can this be done universally, without separate properties for individual annotations? Perhaps in the future we will need other annotations (#76020), which should be resolved in the parser.